### DISCUSSION TITLE
Add horizontal line to a table

### DISCUSSION THREAD

**Discussion by vpoguru** [haozhu233/kableExtra#822] [Q&A] [answered]
I would like to insert a horizontal line above the last row of a table. It is displayed in the RStudio Viewer, but it does not appear in the PDF file. What am I missing?

This is the contents of the R markdown file.

    ---
    output: pdf_document
    ---
    
    ```{r test}
    library(kableExtra)
    
    df <- as.data.frame(matrix(rnorm(20), nrow = 4, ncol = 5))
    
    kable(df) %>%
      kable_classic(full_width = FALSE) %>%
      row_spec(nrow(df), extra_css = "border-top: 1px solid black;")
    ```

**Comment by dmurdoch** [comment]
The CSS code you inserted would only be used in HTML output.  You want PDF output, which is done using LaTeX.  You should use the `hline_after=TRUE` argument on `nrow(df)-1` to get what you want.

**Comment by vpoguru** [comment]
@dmurdoch. That works. But there is quite a large gap after the horizontal line. Can this distance be adjusted?

**Reply by dmurdoch** [reply, marked as answer]
I believe that was fixed in  #816.  Have you tried installing the devel version from Github?  Use
```
remotes::install_github("haozhu233/kableExtra")
```

**Reply by vpoguru** [reply]
Many thanks! Yes, the Github version fixes this issue.
